### ISSUE TITLE
StrBytes / TopicName ergonomics

### ISSUE THREAD

**Issue by bkirwi** [tychedelia/kafka-protocol-rs#43]
Overall I've found the generated message types seem quite easy to work with so far - with the exception of `StrBytes` wrappers like `TopicName`.

From what I can tell, the most straightforward way to construct a `TopicName` from a `String` is:

```rust
let bytes = Bytes::from(string);
let string = <StrBytes as string::TryFrom<Bytes>>::try_from(bytes).expect("bytes from valid string");
string.into()
```

- That's a lot of code for a presumably-very-common operation. (The inverse, getting a `String` from a `TopicName`, is `(**topic).to_owned()`... better but still pretty obscure.)
- This requires me to re-validate the bytes as UTF8, even though I'm constructing it from valid `String` data.
- `TryFrom` is not the standard library implementation of `TryFrom`; it's a custom version in `string`. The issue to fix this has been open for years. This also requires clients to take an explicit dependency on `string` to be able to call the `try_from` method.

A few extra methods on / impls for `TopicName` would probably help with this. The first point looks like it would require calling an `unsafe` method on `string::String`, though, at which point it's probably worth just reimplementing the relevant handful of lines in this codebase and cutting out an unmaintained dep.

**Comment by tychedelia** [comment]
Agree that this is a major point of friction. We've had issues with the `string` crate for years and I've had to bug them to get a simple release done for dependency upgrades in the past. I would happily accept a smaller contribution here that adds helpers, but would also appreciate thoughts on how we might remove this dependency entirely.

**Comment by bkirwi** [comment]
I pushed my thoughts on both questions here: https://github.com/tychedelia/kafka-protocol-rs/pull/44

**Comment by rukai** [comment]
I believe this can be closed due to PR #44
Please reopen if I've misunderstood.
